### ISSUE TITLE
[LSP] - allow lineLength argument on server initialization

### ISSUE THREAD

**Issue by ericdallo** [dart-lang/sdk#43043]
Related to https://github.com/emacs-lsp/lsp-dart/issues/53#issuecomment-673637634

Currently, the VSCode DartCode extension [has an option](https://github.com/Dart-Code/Dart-Code/blob/master/src/extension/config.ts#L169) for specifying the `dart.lineLength` to use when formatting the document.
Is there any way to setup the LSP server to when the client request `textDocument/formatting` the server use the provided `lineLength`?

**Comment by ericdallo** [comment]
c/c @DanTup

**Comment by DanTup** [comment]
@ericdallo in recent builds (that include https://github.com/dart-lang/sdk/commit/ba459f96cfe020e8d6b01f527959e5e1bb900508), this is supported using the `dart.lineLength` workspace setting:

https://github.com/dart-lang/sdk/blob/master/pkg/analysis_server/tool/lsp_spec/README.md#client-workspace-configuration

It's not provided at initialisation, as that means the server must be restarted if the value changes. By using config, it can be modified by the user and take effect immediately.

Let me know if this doesn't work for you. Thanks!

**Comment by ericdallo** [comment]
@DanTup sorry, I didn't notice that addition, this will certainly fix the issue, I'll implement that support on lsp-dart, thank you!

**Comment by ericdallo** [comment]
@DanTup We got [this issue](https://github.com/emacs-lsp/lsp-dart/issues/75) on `lsp-dart`, and after some debugging I could not find why formatting is not working on `lsp-dart`, could you confirm if is everything ok from the server side?
This is the log from `lsp-mode` with the LSP requests/responses:

```javascript
[Trace - 03:06:07 PM] Received request 'workspace/configuration - (1).
Params: {
  "items": [
    {
      "section": "dart"
    }
  ]
}

[Trace - 03:06:07 PM] Sending response 'workspace/configuration - (1)'. Processing request took 0ms
Params: {
  "jsonrpc": "2.0",
  "id": 1,
  "result": [
    {
      "enableSdkFormatter": true,
      "lineLength": 200
    }
  ]
}
```

But the code is still formatted using 80 caracters.

I'm using [this branch](https://github.com/emacs-lsp/lsp-dart/pull/76/files#diff-72f44eb6c0c0717847f9f2db1ff8a5b0R101-R102) for `lsp-dart`. Am I missing something?

**Comment by DanTup** [comment]
The request/response above look correct. Just to clarify - are you saying it works correctly from within VS Code when using `previewLsp`, just not for emacs?

If so, are you able to capture a full log (everything from initialisation, to the bad format response) for me to review? Thanks!

**Comment by ericdallo** [comment]
Yes, for VScode works.

Sample code:
[lsp-dart-bug.zip](https://github.com/dart-lang/sdk/files/5367329/lsp-dart-bug.zip)

LMK if there is some bug with `lsp-dart` or `lsp-mode`

**Comment by DanTup** [comment]
@ericdallo can you confirm which version of the Dart SDK you're using for testing? This setting was included in v2.10.0 of the Dart SDK, but I don't think that's what's used for this log, as it doesn't include [the server info](https://github.com/dart-lang/sdk/blob/2.10.0/pkg/analysis_server/lib/src/lsp/handlers/handler_initialize.dart#L68) which is also included in that version. I think upgrading to the latest SDK may resolve the issue.

**Comment by ericdallo** [comment]
Sure @DanTup, sorry for the delay.
It works using flutter beta channel, dart 2.10.0.
Thank you @DanTup!
